lib/maths/CPeriodicityHypothesisTests.cc

```diff
@@ -69,6 +69,38 @@ const std::size_t MINIMUM_REPEATS_TO_TEST_AMPLITUDE{4};
 //! A high priority for components we want to take precendence.
 double HIGH_PRIORITY{2.0};
 
+//! \brief Used to compute conjunctions of smooth x < t or x > t
+//! conditions.
+class CSmoothCondition {
+public:
+    explicit CSmoothCondition(double value = 0.0) : m_Value{value} {}
+
+    operator bool() const { return m_Value > 1.0; }
+    bool operator<(const CSmoothCondition& rhs) const {
+        return m_Value < rhs.m_Value;
+    }
+
+    double p() const { return m_Value; }
```

Comment:
what does `p` mean? Is this the p-value of a hypothesis test? I would call it `pValue` of `probability` then.

Reply:
It isn't a p-value (in the conventional sense) it is more how close the condition is to passing or not. The code called it p in the past which is why I called it p here, but it isn't a good name. This is really a fuzzy logical operation with AND represented by multiplication (except it there is an implicit threshold for being actually true: a logical operation just holds if all constituent conditions' truth values are 0.5, so the value isn't normalised as it should). Let me come up with something better.

Reply:
I went ahead and renamed some stuff in this area. I think this is better. See [cbf9d37](https://github.com/elastic/ml-cpp/pull/1177/commits/cbf9d3775e8dfd01cc16ed06fd8f6cdbbe4af372).

Reply:
Indeed, stating it explicitly that you are using fuzzy logic makes is much clearer :+1: